### DISCUSSION TITLE
How to use AsyncNotifierProvider to call API only when button is clicked a not when build() method

### DISCUSSION THREAD

**Discussion by Ostreey** [rrousselGit/riverpod#2811] [Q&A]
I'm struggling for a while with how to use riverpod with @riverpod generator to make API call only when button has been pressed but without initial call on build() method. For example i would like to use it to make a call to backend server to log in, therefore i dont want any API calls before user enters credentials. The following code doesnt need credentials but it doesnt matter, im wondering how to do it so initial build() method want make an API call.
<pre>
```
class FbLogin extends _$FbLogin{
  @override
  FutureOr<LoginResult> build() => _getLoginResult();

  Future<void> execute() async{
    state = const AsyncLoading();
    state = await AsyncValue.guard(()async {
      final LoginResult result = await FacebookAuth.instance.login();
      print("Provider inside:: ${result.status}");
      return result;
    });
  }
}
```
</pre>
Thanks in advance
Dawid

**Comment by Velkamhell97** [comment]
Did you find some solution ? I had the same problem.
